src/log/log.c

```diff
@@ -293,47 +292,20 @@ __wt_log_force_sync(WT_SESSION_IMPL *session, WT_LSN *min_lsn)
         __wt_cond_wait(session, log->log_sync_cond, 10000, NULL);
     }
     __wt_spin_lock(session, &log->log_sync_lock);
-    WT_ASSERT(session, log->log_dir_fh != NULL);
+
     /*
      * Sync the directory if the log file entry hasn't been written into the directory.
      */
-    if (log->sync_dir_lsn.l.file < min_lsn->l.file) {
-        __wt_verbose(session, WT_VERB_LOG,
-          "log_force_sync: sync directory %s to LSN %" PRIu32 "/%" PRIu32, log->log_dir_fh->name,
-          min_lsn->l.file, min_lsn->l.offset);
-        time_start = __wt_clock(session);
-        WT_ERR(__wt_fsync(session, log->log_dir_fh, true));
-        time_stop = __wt_clock(session);
-        fsync_duration_usecs = WT_CLOCKDIFF_US(time_stop, time_start);
-        WT_ASSIGN_LSN(&log->sync_dir_lsn, min_lsn);
-        WT_STAT_CONN_INCR(session, log_sync_dir);
-        WT_STAT_CONN_INCRV(session, log_sync_dir_duration, fsync_duration_usecs);
-    }
+    WT_ERR(__log_fsync_dir(session, min_lsn, "log_force_sync"));
+
     /*
-     * Sync the log file if needed.
+     * Sync the log file if needed. Get our own file handle to the log file. It is possible for the
```

Comment:
I think the comment needs to go with the code movement. It doesn't make sense here because this code is not getting a file handle to the log file anymore.

After going through the whole PR I see why this is here. If you want to keep it here make it clear that "Get our own file handle..." means sending `true` in to the `log_fsync_file` call.

Reply:
Done.